flow/project.py

```diff
@@ -5092,15 +5064,7 @@ def _show_traceback_and_exit(error):
             if is_user_error:
                 # Always show the user traceback cause.
                 error = error.__cause__
```

Comment:
This part I'm not sure about. Is this hiding the internal traceback by setting `error = error.__cause__`? This is supposed to catch cases where signac-flow fails _because_ of failures in user code, and then it hides the signac-flow exception in favor of the exception in user code that signac-flow caught. We should probably make this consistent with the deleted code above:

https://github.com/glotzerlab/signac-flow/pull/674/files#diff-d10cda9deccafc1b555d818a01e3082c56b961a4d6e837eb4a65cb87d4b3ca46L4718-L4720

Reply:
The options are:
1. Show the original `error`, which makes it look like signac-flow is failing despite the exception coming from user code (users have to scroll up a ways to see their error)
2. If it's an error that we can identify as coming from user code, always show `error.__cause__` instead

We should do the same thing here and in the code that was deleted above.

Reply:
Hmm... Good point. I say we just show the full stack then and raise the original error.